File: 0816-design-hashset/0816-design-hashset.py
class MyHashSet:

    class _Node:
        def __init__(self, key, next=None):
            self.key = key
            self.next = next

    def __init__(self):
        self._slots = [MyHashSet._Node(0) for _ in range(1000)]
        self.size = len(self._slots)


    def _hash_function(self, key: int) -> int:
        return key % self.size

    def add(self, key: int) -> None:
        index = self._hash_function(key)
        head = self._slots[index]
        while head.next:
            if head.next.key == key:
                return
            head = head.next
        head.next = MyHashSet._Node(key)

    def remove(self, key: int) -> None:
        index = self._hash_function(key)
        head = self._slots[index]
        while head.next:
            if head.next.key == key:
                head.next = head.next.next
                return
            head = head.next
            

    def contains(self, key: int) -> bool:
        index = self._hash_function(key)
        head = self._slots[index]
        while head.next:
            if head.next.key == key:
                return True
            head = head.next
        return False

# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)

File: 0816-design-hashset/0816-design-hashset.java
class MyHashSet {

    // Nested Class that represents an LinkedList
    private static class Node {
        int value;
        Node next;

        public Node(int val) {
            this.value = val;
        }

        public int getValue() {
            return this.value;
        }

        public Node getNext() {
            return this.next;
        }
    }

    private final Node[] set; // Array of Nodes 

    public MyHashSet() {
        set = new Node[10000];
        for (int i = 0; i < set.length; i++) {
            set[i] = new Node(0);
        }
    }
    
    public void add(int key) {
        Node current = set[key % set.length];
        while (current.next != null) {
            if (current.next.value == key) return;
            current = current.next;
        }

        current.next = new Node(key);
    }
    
    public void remove(int key) {
        Node current = set[key % set.length];

        while (current.next != null) {
            if (current.next.value == key) {
                current.next = current.next.next;
                return;
            }

            current = current.next;
        }
    }
    
    public boolean contains(int key) {
        Node current = set[key % set.length];

        while (current.next != null) {
            if (current.next.value == key) return true;
            current = current.next;
        }

        return false;
    }
}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet obj = new MyHashSet();
 * obj.add(key);
 * obj.remove(key);
 * boolean param_3 = obj.contains(key);
 */